src/sentry/api/serializers/models/integration.py

```diff
@@ -58,7 +65,7 @@ def __init__(
 
     def serialize(
         self,
-        obj: Integration,
+        obj: Integration | APIIntegration,
```

Comment:
This all really, really makes me sad.  Is there no way to avoid serializing APIIntegration objects?

Preferably, serializers always work on ORM objects, and call service methods to serialize related objects that exist in a far off silo.  For instance, `user_service.serialize_users` allows region silos to "serialize" users without making the unfortunate complication of having Serializer classes actually, serialize APIUsers (Which would devolve quickly into pathological ugliness).

Is there, any way we can revert this and say, just implement an integration_service.serialize_integrations call which invokes these in the "aligned" silo?

Reply:
@corps  Encountering some issues with this since the serializers themselves can't safely be called by the Control Silo (they use Region Silo models in `get_attrs`. Do you have any ideas for how to avoid this? 

Context: https://github.com/getsentry/sentry/blob/master/src/sentry/api/serializers/models/integration.py#L202-L214

I can't have the service accept the `Group` instance, so I have to pass an ID, but the serializer expects the actual `Group` instance, which is not available in the control silo.

Reply:
Addressed, pinned the type at APIIntegration and moved the serializers.